charts/kube-otel-stack/templates/autoinstrumentation.yaml

```diff
@@ -1,23 +1,33 @@
-{{ if .Values.autoinstrumentation.enabled }}
+{{- if .Values.autoinstrumentation.enabled }}
 apiVersion: opentelemetry.io/v1alpha1
 kind: Instrumentation
 metadata:
   name: {{ $.Release.Name }}-instrumentation
 spec:
   exporter:
     endpoint: http://{{ $.Release.Name }}-{{ .Values.autoinstrumentation.collectorTarget }}-collector.{{ $.Release.Namespace }}:4317
-  propagators:
-{{ toYaml .Values.autoinstrumentation.propagators | indent 4 }}
-  {{ with .Values.autoinstrumentation.sampler }}
-  sampler:
-{{ toYaml . | indent 4 }}
-  {{ end }}
-  {{ with .Values.autoinstrumentation.env }}
-  env:
-{{ toYaml . | indent 4 }}
-  {{ end }}
-  {{ with .Values.autoinstrumentation.resource }}
-  resource:
-{{ toYaml . | indent 4 }}
-  {{ end }}
+  propagators: {{ toYaml .Values.autoinstrumentation.propagators | nindent 4 }}
+  {{- with .Values.autoinstrumentation.sampler }}
+  sampler: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.autoinstrumentation.env }}
+  env: {{ toYaml . | nindent 4 }}
```

Comment:
Its usually better to have the `toYaml` line in its own line with `{{-` while using `nindent` so you can both visualize the rendered template while making sure there's no extra spaces.